homeassistant/helpers/config_validation.py

```diff
@@ -1790,6 +1790,8 @@ def _backward_compat_trigger_schema(value: Any | None) -> Any:
             )
         value = dict(value)
         value[CONF_PLATFORM] = value.pop(CONF_TRIGGER)
+    elif CONF_PLATFORM not in value:
+        raise vol.Invalid("required key not provided", ["trigger"])
```

Comment:
It should be like this, sorry for the bad suggestion
```suggestion
        raise vol.Invalid("required key not provided", [CONF_TRIGGER])
```